### PR TITLE
Add B2B methods

### PR DESCRIPTION
**Pull request by logan-stytch**
ruby-sample

### CODE REVIEW

File: lib/stytch.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-require_relative 'stytch/client'
-require_relative 'stytch/middleware'
-require_relative 'stytch/version'
+require_relative "stytch/b2b_client"
+require_relative "stytch/client"
+require_relative "stytch/middleware"
+require_relative "stytch/version"
 
 module Stytch end
+module StytchB2B end
```
Comment (logan-stytch) [inline lib/stytch.rb:12]: auto-formatter preferred double quotes

Other question -- do I have to declare StytchB2B like this?

File: lib/stytch/b2b_client.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      when /\Ahttps?:\/\//
+        # If this is a string that looks like a URL, assume it's an internal development URL.
+        env
+      else
+        if project_id.start_with? "project-live-"
+          "https://api.stytch.com"
+        else
+          "https://test.stytch.com"
+        end
+      end
```
Comment (logan-stytch) [inline lib/stytch/b2b_client.rb:48]: Now featuring env detection, but I'm curious if there's a cleaner way to do this

File: lib/stytch/debug.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+require_relative "request_helper"
+
+module Stytch
```
Comment (max-stytch) [inline lib/stytch/debug.rb:5]: We haven't made the debug endpoint public before, have we?

File: lib/stytch/client.rb [hunk 4]
```diff
@@ -46,7 +46,11 @@ def api_host(env)
         # If this is a string that looks like a URL, assume it's an internal development URL.
         env
       else
-        raise ArgumentError, "Invalid value for env (#{env}): should be :live or :test"
+        if project_id.start_with? 'project-live-'
```
Comment (max-stytch) [inline lib/stytch/client.rb:49]: <3

File: lib/stytch/b2b_sso.rb [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      @saml = StytchB2B::SSO::SAML.new(@connection)
+    end
+
+    def get_connections(
+      organization_id:
+    )
+      query_params = {
+        organization_id: organization_id
+      }
+      request = request_with_query_params("/v1/b2b/sso/#{organization_id}", query_params)
```
Comment (max-stytch) [inline lib/stytch/b2b_sso.rb:23]: Weird to see this as a query param - I think of it as a path param

File: lib/stytch/b2b_sso.rb [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    class OIDC
+      include Stytch::RequestHelper
+
+      def initialize(connection)
+        @connection = connection
+      end
+
+      def create_connection(
+        organization_id:,
+        display_name:
```
Comment (max-stytch) [inline lib/stytch/b2b_sso.rb:63]: Display name is optional

File: lib/stytch/b2b_sso.rb [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+          display_name: display_name
+        }
+
+        post_request("/v1/b2b/sso/oidc/#{organization_id}", request)
+      end
+
+      def update_connection(
+        organization_id:,
+        connection_id:,
+        display_name:,
```
Comment (max-stytch) [inline lib/stytch/b2b_sso.rb:76]: All fields other than OrgId and ConnId are optional - we default to (and ignore) the empty string

File: lib/stytch/b2b_sso.rb [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        request[:idp_entity_id] = idp_entity_id unless idp_entity_id.nil?
+        request[:display_name] = display_name unless display_name.nil?
+        request[:attribute_mapping] = attribute_mapping unless attribute_mapping.nil?
+        request[:x509_certificate] = x509_certificate unless x509_certificate.nil?
+        request[:idp_sso_url] = idp_sso_url unless idp_sso_url.nil?
+
+        put_request("/v1/b2b/sso/saml/#{organization_id}/connections/#{connection_id}", request)
+      end
+
+      def url(
```
Comment (max-stytch) [inline lib/stytch/b2b_sso.rb:143]: This endpoint is not yet implemented

File: lib/stytch/b2b_sso.rb [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+      )
+        request = {
+          connection_id: connection_id,
+          metadata_url: metadata_url
+        }
+
+        put_request("/v1/b2b/sso/saml/#{connection_id}/url", request)
+      end
+
+      def doc(
```
Comment (max-stytch) [inline lib/stytch/b2b_sso.rb:155]: Not yet implemented

File: lib/stytch/b2b_passwords.rb [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+
+      def initialize(connection)
+        @connection = connection
+      end
+
+      def reset_start(
+        organization_id:,
+        email_address:,
+        reset_password_redirect_url:,
+        login_redirect_url:,
```
Comment (max-stytch) [inline lib/stytch/b2b_passwords.rb:93]: IIRC we default to the default if this is not provided

File: lib/stytch/b2b_passwords.rb [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+
+      post_request('/v1/b2b/passwords/strength_check', request)
+    end
+
+    def migrate(
+      email_address:,
+      hash:,
+      hash_type:,
+      organization_id:,
+      name:,
```
Comment (max-stytch) [inline lib/stytch/b2b_passwords.rb:35]: not required

File: lib/stytch/b2b_organizations.rb [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+
+      def organizations_delete(
+        organization_id:,
+        member_id:
+      )
+        delete_request("/v1/b2b/organizations/#{organization_id}/members/#{member_id}")
+      end
+
+      def search(
+        cursor:,
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:182]: not required

File: lib/stytch/b2b_organizations.rb [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+    end
+
+    class Members
+      include Stytch::RequestHelper
+
+      def initialize(connection)
+        @connection = connection
+      end
+
+      def organizations_update(
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:154]: This should just be called `update`

File: lib/stytch/b2b_organizations.rb [hunk 14]
```diff
@@ -0,0 +1,205 @@
...
+      request = {
+        cursor: cursor
+      }
+      request[:limit] = limit unless limit.nil?
+      request[:query] = query unless query.nil?
+
+      post_request('/v1/b2b/organizations/search', request)
+    end
+
+    def member(
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:133]: This should be called `get` and be on the `Member` class

File: lib/stytch/b2b_organizations.rb [hunk 15]
```diff
@@ -0,0 +1,205 @@
...
+      put_request("/v1/b2b/organizations/#{organization_id}", request)
+    end
+
+    def delete(
+      organization_id:
+    )
+      delete_request("/v1/b2b/organizations/#{organization_id}")
+    end
+
+    def member_create(
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:97]: Probably should be called `create` and live on `Member`

File: lib/stytch/b2b_organizations.rb [hunk 16]
```diff
@@ -0,0 +1,205 @@
...
+      organization_id:,
+      email_address:,
+      create_member_as_pending:,
+      is_breakglass:,
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:101]: not required

File: lib/stytch/b2b_organizations.rb [hunk 17]
```diff
@@ -0,0 +1,205 @@
...
+
+      @members = StytchB2B::Organizations::Members.new(@connection)
+    end
+
+    def create(
+      organization_name:,
+      organization_slug:,
+      organization_logo_url:,
+      email_allowed_domains:,
+      allowed_auth_methods:,
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:21]: not required

File: lib/stytch/b2b_organizations.rb [hunk 18]
```diff
@@ -0,0 +1,205 @@
...
+      @connection = connection
+
+      @members = StytchB2B::Organizations::Members.new(@connection)
+    end
+
+    def create(
+      organization_name:,
+      organization_slug:,
+      organization_logo_url:,
+      email_allowed_domains:,
```
Comment (max-stytch) [inline lib/stytch/b2b_organizations.rb:20]: not required

**Review by max-stytch — APPROVED** [review]

**Pull request merged**